### PR TITLE
Add compatibility with PHP 8.1

### PR DESCRIPTION
**Pull request by pascalheidmann**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description
This PR updates some dependencies to be compatible with PHP 8.1. Also adds minor improvements to tests like type hints & return types.

### Notice
It uses a reference to https://github.com/laminas/laminas-i18n/pull/62 for laminas-i18n which should be merged and replaced with a real release before this PR can be merged.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -64,5 +65,11 @@
     },
     "replace": {
         "zendframework/zend-expressive-zendrouter": "^3.0.1"
-    }
+    },
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "https://github.com/cedric-anne/laminas-i18n.git"
```
Comment (Ocramius) [inline composer.json]: This should not be part of the patch - it should already be handled by `.laminas-ci.json`

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (froschdesign) [reply composer.json]: @pascalheidmann 
laminas-i18n with support for PHP 8.1 is [released](https://github.com/laminas/laminas-i18n/releases/tag/2.12.0), so you can remove this part.

File: .laminas-ci.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "ignore_php_platform_requirements": {
+    "8.1": true
```
Comment (internalsystemerror) [inline .laminas-ci.json:3]: Is this still needed with the new dependency releases?

**Review by internalsystemerror — CHANGES_REQUESTED** [review]
The files added to .gitignore also need to be removed from this commit (.phpcs-cache and .phpunit.result.cache).

File: .laminas-ci.json (hunk 2 above)
Comment (Ocramius) [reply .laminas-ci.json:3]: Let's keep it for now

**Review by Ocramius — APPROVED** [review]
LGTM, thanks @pascalheidmann

**Pull request merged**